src/gui/accountstate.cpp

```diff
@@ -550,6 +554,28 @@ void AccountState::slotCheckConnection()
     }
 }
 
+void AccountState::slotCheckServerAvailibility()
+{
+    if (state() == AccountState::Connected
+        || state() == AccountState::SignedOut
+        || state() == AccountState::MaintenanceMode
+        || state() == AccountState::AskingCredentials) {
+        qCInfo(lcAccountState) << "Skipping server availibility check for account" << _account->davUser() << "with state" << state();
+        return;
+    }
+    qCInfo(lcAccountState) << "Checking server availibility for account" << _account->davUser();
+    const auto serverAvailibilityUrl = Utility::concatUrlPath(_account->url(), QLatin1String("/index.php/204"));
+    auto checkServerAvailibilityJob = _account->sendRequest(QByteArrayLiteral("GET"), serverAvailibilityUrl);
+    connect(checkServerAvailibilityJob, &SimpleNetworkJob::finishedSignal, this, [this](QNetworkReply *reply) {
+        if (reply->attribute(QNetworkRequest::HttpStatusCodeAttribute) == 204) {
+            qCInfo(lcAccountState) << "Server is now available for account" << _account->davUser();
+            _lastCheckConnectionTimer.invalidate();
+            resetRetryCount();
+            QMetaObject::invokeMethod(this, "slotCheckConnection", Qt::QueuedConnection);
```

Comment:
can you use the method pointer syntax ?
that is needed to be able to compile check this call